src/kafka-extensions/api.ts

```diff
@@ -0,0 +1,31 @@
+import { Kafka } from "kafkajs";
+import { Cluster, ConnectionOptions } from "../client/client";
+import { ClusterSettings } from "../settings/clusters";
+
+export interface ClusterProviderProcessorProvider {
+
+    getClusterProviderProcessor(clusterProviderId: string) : ClusterProviderProcessor;
+
+}
+
+/**
+ * The cluster provider processor.
+ */
+export interface ClusterProviderProcessor {
+
+    /**
+     * Returns the clusters managed by the processor which must be added to the kafka explorer.
+     *
+     * @param clusterSettings the cluster settings.
+     */
+    collectClusters(clusterSettings: ClusterSettings): Promise<Cluster[] | undefined>;
+
+    /**
+     * Create the Kafka JS client instance fromthe given connection options.
+     * When processor doesn't implement this method, the Kafka JS client
+     * instance is created with the default client factory from the vscode-kafka.
+     *
+     * @param connectionOptions the connection options.
+     */
+    createKafka?(connectionOptions: ConnectionOptions): Kafka;

```

Comment:
It might be better to decouple the api from kafkajs, and instead require an object with the same params than a kafka instance

Reply:
What about having 

`createKafkaConfig?(connectionOptions: ConnectionOptions): KafkaConfig;`

External extension could depends on kafkajs to have the proper KafkaConfig interface structure or it could just redeclare this interface in extension.ts.

What do you think about that?

Reply:
Yes I was thinking about a KafkaConfig object like that

Reply:
fixed